src/frontend/lib/security.ts

```diff
@@ -22,3 +30,32 @@ export function getLockoutThreshold(attempts: number): number | null {
 
   return 0;
 }
+
+export function generateSalt(): string {
+  const bytes = Crypto.getRandomBytes(8);
+  return [...bytes].map(b => b.toString(16).padStart(2, '0')).join('');
+}
+
+export async function hashPasscode(
+  passcode: string,
+  salt: string,
+): Promise<string> {
+  const toHash = `${salt}:${passcode}`;
+  const hash = await Crypto.digestStringAsync(
+    Crypto.CryptoDigestAlgorithm.SHA256,
+    toHash,
+  );
+  return `${salt}:${hash}`;
+}
+
+export async function verifyPasscode(
+  input: string,
+  stored: string,
+): Promise<boolean> {
```

Comment:
When there are 2 or more parameters, we should use an object so it is less ambigous. Eg:

```ts
hashPasscode({passcode: string,
salt: string})

verifyPasscode({input:string, stored:string})


```